internal/checks/blockheight.go

```diff
@@ -175,7 +177,7 @@ func (c *BlockHeightCheck) subscribeNewHead() error {
 		c.setError(c.webSocketError)
 	}
 
-	wsClient, err := c.clientGetter(c.upstreamConfig.WSURL, &client.BasicAuthCredentials{Username: c.upstreamConfig.BasicAuthConfig.Username, Password: c.upstreamConfig.BasicAuthConfig.Password})
+	wsClient, err := c.clientGetter(c.upstreamConfig.WSURL, &c.upstreamConfig.BasicAuthConfig, &c.upstreamConfig.RequestHeadersConfig)
```

Comment:
Super nit: Might be better to pass this by value instead of passing the pointer. 

I also dont't think you need to check
`if additionalRequestHeaders == nil {
		return
	}`

in `ethereum_client` either since [] == nil

Reply:
Good call on `[] == nil`!

> Super nit: Might be better to pass this by value instead of passing the pointer.

Mind explaining why?

Reply:
Gonna merge for now! Happy to update in a follow up PR.

Reply:
It's mainly cause I don't really see a reason to pass it by pointer since nil == []. Also you need to deference the pointer to access the underlying slice, which is clunkiness that can be avoided.